contracts/modules/TransactionManager.sol

```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * @notice Removes an address from the whitelist of a wallet.
+     * @param _wallet The target wallet.
+     * @param _target The address to remove.
+     */
+    function removeFromWhitelist(
+        address _wallet,
+        address _target
+    )
+        external
+        onlyWalletOwnerOrSelf(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        setWhitelist(_wallet, _target, 0);
+        emit RemovedFromWhitelist(_wallet, _target);
+    }
+
+    function toggleDappRegistry(
+        address _wallet,
+        bytes32 _registry
+    )   
+        external
+        onlySelf()
+        onlyWhenUnlocked(_wallet)
+    {
+        bool isEnabled = authoriser.toggle(_wallet, _registry);
+        emit ToggledDappRegistry(_wallet, _registry, isEnabled);
+    }
+
+    /**
+    * @notice Checks if an address is whitelisted for a wallet.
+    * @param _wallet The target wallet.
+    * @param _target The address.
+    * @return _isWhitelisted true if the address is whitelisted.
+    */
+    function isWhitelisted(
+        address _wallet,
+        address _target
+    )
+        public
+        view
+        returns (bool _isWhitelisted)
+    {
+        uint whitelistAfter = userWhitelist.getWhitelist(_wallet, _target);
+        return whitelistAfter > 0 && whitelistAfter < block.timestamp;
+    }
+
+    /** ******************* Callbacks ************************** */
+
+    /**
+    * @notice Implementation of EIP 1271.
+    * Should return whether the signature provided is valid for the provided data.
+    * @param _msgHash Hash of a message signed on the behalf of address(this)
+    * @param _signature Signature byte array associated with _msgHash
+    */
+    function isValidSignature(
+        bytes32 _msgHash,
+        bytes memory _signature
+    ) 
+        external
+        view
+        returns (bytes4)
+    {
+        require(_signature.length == 65, "TM: invalid signature length");
+        address signer = Utils.recoverSigner(_msgHash, _signature, 0);
+        require(_isOwner(msg.sender, signer), "TM: Invalid signer");
+        return ERC1271_ISVALIDSIGNATURE;
+    }
+
+    /**
+     * @notice Implementation of ERC721
+     * @notice An ERC721 smart contract calls this function on the recipient contract
+     * after a `safeTransfer`. If the recipient is a BaseWallet, the call to onERC721Received
+     * will be forwarded to the method onERC721Received of the present module.
+     * @return bytes4 `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+     */
+    function onERC721Received(
+        address /* operator */,
+        address /* from */,
+        uint256 /* tokenId */,
+        bytes calldata /* data*/
+    )
+        external
+        returns (bytes4)
+    {
+        return ERC721_RECEIVED;
+    }
+
+    // *************** Internal Functions ********************* //
+
+    function enableStaticCall(address _wallet) internal {
```

Comment:
Reminder to add a method that the owner can call to enable a static call implemented in an external contract and executed via this module's fallback.

Reply:
Maybe the fallback could be implemented by reading an entry in a `mapping(bytes4 => StaticCallExecution)` where `StaticCallExecution` is a 
```
struct {
    bytes4 result;
    address implementation;
}
``` 
where in most cases `implementation` is `address(0)` in which case `result` is simply returned by the fallback (e.g. for  `onERC721Received` and `onERC1155Received`) but in some special cases `result` is null and `implementation` contains an implementing contract (as would be the case for `isValidSignature`).

Reply:
This will also allow us to save a bunch of bytecode by not having to store any specific static call logic in the TransactionManager.

Reply:
Implemented in #195